test/e2e/helm/kuma_helm_upgrade_multizone.go

```diff
@@ -0,0 +1,154 @@
+			Install(Kuma(core.Global,
+				WithInstallationMode(HelmInstallationMode),
+				WithHelmChartPath(Config.HelmChartName),
+				WithHelmReleaseName(releaseName),
+				WithHelmChartVersion(oldestSupportedVersion),
+				WithoutHelmOpt("global.image.tag"),
+			)).
+			Setup(global)
+		Expect(err).ToNot(HaveOccurred())
+
+		globalCP = global.GetKuma()
+		Expect(globalCP).ToNot(BeNil())
+	})
+
+	It("should install a Kuma version 2 minor releases behind the current version on Zone", func() {
+		err := NewClusterSetup().
+			Install(Kuma(core.Zone,
+				WithInstallationMode(HelmInstallationMode),
+				WithHelmChartPath(Config.HelmChartName),
+				WithHelmReleaseName(releaseName),
+				WithHelmChartVersion(oldestSupportedVersion),
+				WithGlobalAddress(globalCP.GetKDSServerAddress()),
+				WithHelmOpt("ingress.enabled", "true"),
+				WithoutHelmOpt("global.image.tag"),
+			)).
+			Setup(zone)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	numberOfPolicies := func(c Cluster) (int, error) {
+		output, err := c.GetKumactlOptions().RunKumactlAndGetOutput("get", "meshtimeouts", "-m", "default", "-o", "json")
+		if err != nil {
+			return 0, err
+		}
+		t := struct {
+			Total int `json:"total"`
+		}{}
+		if err := json.Unmarshal([]byte(output), &t); err != nil {
+			return 0, err
+		}
+		return t.Total, nil
+	}
+
+	It("should sync policies from Global to Zone", func() {
+		// when apply policy on Global
+		Expect(YamlK8s(fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: mt1
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        idleTimeout: 20s
+        http:
+          requestTimeout: 2s
+          maxStreamDuration: 20s`, Config.KumaNamespace, "default"))(global)).To(Succeed())
+
+		// then the policy is synced to Zone
+		Eventually(func(g Gomega) int {
+			n, err := numberOfPolicies(zone)
+			g.Expect(err).ToNot(HaveOccurred())
+			return n
+		}, "30s", "1s").Should(Equal(1))
+	})
+
+	It("should upgrade Kuma on Global", func() {
+		err := global.(*K8sCluster).UpgradeKuma(core.Global,
+			WithHelmReleaseName(releaseName),
+			WithHelmChartPath(Config.HelmChartPath),
+			ClearNoHelmOpts(),
+		)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	// Disabled due to https://github.com/kumahq/kuma/issues/9184
+	PIt("should upgrade Kuma on Zone", func() {
+		// when
+		err := zone.(*K8sCluster).UpgradeKuma(core.Zone,
+			WithHelmReleaseName(releaseName),
+			WithHelmChartPath(Config.HelmChartPath),
+			ClearNoHelmOpts(),
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		// then
+		Consistently(func(g Gomega) int {
+			n, err := numberOfPolicies(global)
+			g.Expect(err).ToNot(HaveOccurred())
+			return n
+		}, "10s", "100ms").Should(Equal(1))
+		// and
+		Consistently(func(g Gomega) int {
+			n, err := numberOfPolicies(zone)
+			g.Expect(err).ToNot(HaveOccurred())
+			return n
+		}, "10s", "100ms").Should(Equal(1))
```

Comment:
nit: can we squeeze this to one consistently? 20s overhead is a lot :( Should this be 5s?

Reply:
yeah, can be squeezed into 1 Consistently. Unfortunately, 5s is not enough to track the issue, but maybe I can wait for a new subscription on ZoneInsight with Eventually and then do a 5s Consistently that checks the number of policies

Reply:
Added Eventually that waits for the new ZoneInsight subscription. So now I was able to create 1 Consistently with 5s.